lib/terraform/runner.rb

```diff
@@ -180,6 +187,26 @@ def encoded_zip_from_directory(template_path)
         end
       end
 
+      # Parse Variables in Terraform Template
+      def template_variables(
+        template_path
+      )
+        _log.info("prase template: #{template_path}")
+        encoded_zip_file = encoded_zip_from_directory(template_path)
+
+        # TODO: use tags,env_vars
+        payload = {
+          :templateZipFile => encoded_zip_file,
+        }
+
+        http_response = terraform_runner_client.post(
+          "api/template/variables",
+          *json_post_arguments(payload)
+        )
```

Comment:
Interesting I thought parsing the template variables was something we could do entirely from the committed files, didn't realize it was an API call to the runner

Reply:
The best way to parse terraform template is to use hashicorp's own tool => [terraform-config-inspect](https://github.com/hashicorp/terraform-config-inspect).  

The same is already used in CAM,  so using the api.